src/rules/azureResourceGraph.ts

```diff
@@ -77,35 +108,161 @@ export class ResourceGraphRule implements BaseRule<ResourceGraphTarget> {
   getQueryByGroups(groupNames: string[]) {
     const formattedGroups = groupNames.map(name => `'${name}'`).join(', ');
     const groupQuery = `| where resourceGroup in~ (${formattedGroups})`;
-    const firstPipeIndex = this.query.indexOf('|');
+    const firstPipeIndex = this.evaluation.query.indexOf('|');
     if (firstPipeIndex === -1) {
       // while it is a Microsoft recommendation to start Resource Graph queries with the table name,
       // for this application it is a requirement in order to support filtering for resource groups in the query
       throw Error("Invalid Query. All queries must start with '<tableName> |'");
     } else {
-      const initalTable = this.query.slice(0, firstPipeIndex - 1);
-      const queryEnding = this.query.slice(firstPipeIndex);
+      const initalTable = this.evaluation.query.slice(0, firstPipeIndex - 1);
+      const queryEnding = this.evaluation.query.slice(firstPipeIndex);
       return `${initalTable} ${groupQuery} ${queryEnding}`;
     }
   }
 
-  toScanResult(response: ResourceGraphModels.ResourcesResponse): ScanResult {
+  async sendRequest(
+    target: ResourceGraphTarget,
+    resourceId: string,
+    request: RequestEvaluationObject
+  ) {
+    if (!isRequestEvaluation(this.evaluation)) {
+      throw Error('A valid request evaluation was not found');
+    }
+    const token = await target.credential.getToken(
+      'https://graph.microsoft.com/.default'
+    );
+    const resourceManagementClient = await this.getResourceManagmentClient(
+      resourceId
+    );
+    const options = {
+      url: await this.getRequestUrl(
+        resourceId,
+        resourceManagementClient,
+        request
+      ),
+      method: request.httpMethod as HttpMethods,
+      headers: {
+        Authorization: `Bearer ${token?.token}`,
+        'Content-Type': 'application/json',
+      },
+    };
+    let response = await resourceManagementClient.sendRequest(options);
+    // if the response returns an error because the default api verison is invalid, then parse the error message to retrieve a valid one and try again
```

Comment:
This approach might work (and indeed be faster?), but it feels maybe like a hacky version of calling against a known-good api-version, which are available via the API here: https://docs.microsoft.com/en-us/rest/api/resources/providers/list

I'm sure there's an SDK equivalent somewhere. Not sure it's worth the time to change this right away - something for the backlog